### PR TITLE
add support to retrieve available Carrier Types

### PR DESCRIPTION
**Pull request by jchen293**
This PR adds support for users to retrieve available Carrier Types.

### CODE REVIEW

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Please add a description to each function's docstring (see code in other classes for an example).

**Review by jontsai — APPROVED** [review]

File: src/main/java/com/easypost/model/CarrierType.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     * Set the type of carrier.
+     *
+     * @param type type of a carrier.
+     */
+    public void setType(final String type) {
+        this.type = type;
+    }
+
+    /**
+     * Get the readable of the carrier.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:41]:
```suggestion
     * Get the readable name of the carrier.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @return return readable from a carrier.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:43]:
```suggestion
     * @return return readable name of the carrier.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public String getReadable() {
+        return readable;
+    }
+
+    /**
+     * Set the readable of carrier.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:50]:
```suggestion
     * Set the readable name of the carrier.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @param readable readable for a carrier.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:52]:
```suggestion
     * @param readable readable name of the carrier.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+     * Set the fields of carrier.
+     *
+     * @param fields fields for a carrier.
+     */
+    public void setFields(final Map<String, Object> fields) {
+        this.fields = fields;
+    }
+
+    /**
+     * Retrieve a list of available carrier for the account.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:95]:
```suggestion
     * Retrieve a list of available carriers for the account.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @return list of carrier type that is available to the account.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:97]:
```suggestion
     * @return list of carrier types that are available to the account.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+     * @throws EasyPostException when the request fails.
+     */
+    public static List<CarrierType> all() throws EasyPostException {
+        return all(null);
+    }
+
+    /**
+     * Retrieve a list of available carrier for the given account API key.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:105]:
```suggestion
     * Retrieve a list of available carriers for the given account.
```

File: src/main/java/com/easypost/model/CarrierType.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @param apikey the API key from the user input.
+     * @return list of carrier type that is available to the account of the given API key.
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:108]:
```suggestion
     * @return list of carrier types that are available for the given account.
```

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Code looks fine, you'll want to try and find a consistent lexicon of phrases and use throughout the whole PR. I see a lot of switching between `of`, `for` and `a`, `the` when referencing the carrier accounts. Similarly there are some inconsistencies with pluralism that should be corrected.

We want to try being as consistent and clean as possible 🙂

**Comment by nwithan8** [comment]
> Code looks fine, you'll want to try and find a consistent lexicon of phrases and use throughout the whole PR. I see a lot of switching between `of`, `for` and `a`, `the` when referencing the carrier accounts. Similarly there are some inconsistencies with pluralism that should be corrected.
> 
> We want to try being as consistent and clean as possible 🙂

Regarding this, if it's a static function, it would be getting `a/an` object or something relative to `a/an` object.
If it's a non-static function, relying on an instance of an object, then it's getting `the` object or something relative to `the` object.

**Review by nwithan8 — APPROVED** [review]

File: src/main/java/com/easypost/model/CarrierType.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * Retrieve a list of available carriers for the given account.
+     *
+     * @param apikey the API key from the user input.
+     * @return list of carrier types that are available for the given account.
+     * @throws EasyPostException when the request fails.
+     */
+    public static List<CarrierType> all(final String apikey) throws EasyPostException {
+        CarrierType[] ret = request(RequestMethod.GET, classURL(CarrierType.class), null, CarrierType[].class, apikey);
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/CarrierType.java:112]: Nit: let's call `ret` what it actually is which is `response`. Won't block on it.

**Review by Justintime50 — DISMISSED** [review]

File: src/main/java/com/easypost/model/CarrierType.java (hunk 9 above)
Comment (jchen293) [reply src/main/java/com/easypost/model/CarrierType.java]: resolved.

**Review by Justintime50 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**